src/mgmt/srv_container.c

```diff
@@ -36,13 +36,18 @@ ds_mgmt_cont_set_owner(uuid_t pool_uuid, d_rank_list_t *svc_ranks,
 	daos_prop_t	*prop;
 	uint32_t	prop_nr = 0;
 	uint32_t	i = 0;
+	bool		user_set = false;
+	bool		grp_set = false;
 
 	D_DEBUG(DB_MGMT, "Setting owner for container "DF_UUID" in pool "
 		DF_UUID"\n", DP_UUID(cont_uuid), DP_UUID(pool_uuid));
 
-	if (user != NULL)
+	user_set = user != NULL && strnlen(user, DAOS_ACL_MAX_PRINCIPAL_LEN) > 0;
```

Comment:
If the user string parameter could be malformed, something such as this could be more adapted.
```suggestion
	user_set = user != NULL && strnlen(user, DAOS_ACL_MAX_PRINCIPAL_LEN) > 0 && strnlen(user, DAOS_ACL_MAX_PRINCIPAL_LEN) < DAOS_ACL_MAX_PRINCIPAL_LEN;
```
If my remark is relevant, it should also be applied to the test of the next line (i.e. line 46).